src/Microsoft.Azure.WebJobs.Host/Bindings/ConverterManagerExtensions.cs

```diff
@@ -0,0 +1,234 @@
+                {
+                    continue;
+                }
+
+                var parameters = method.GetParameters();
+                if (!CheckArg(parameters[0].ParameterType, typeSource, genericArgs))
+                {
+                    continue;
+                }
+
+                // Possible match 
+                var typeArgs = typeConverter.GetGenericArguments();
+                int len = typeArgs.Length;
+                var actualTypeArgs = new Type[len];
+                for (int i = 0; i < len; i++)
+                {
+                    actualTypeArgs[i] = genericArgs[typeArgs[i].Name];
+                }
+
+                Type finalType = typeConverter;
+                if (typeConverter.IsGenericTypeDefinition)
+                {
+                    finalType = typeConverter.MakeGenericType(actualTypeArgs);
+                }
+
+                try
+                {
+                    var instance = Activator.CreateInstance(finalType, constructorArgs);
+                    var resolvedMethod = Resolve(finalType, method);
+                    return Tuple.Create(instance, resolvedMethod);
+                }
+                catch (TargetInvocationException e)
+                {
+                    throw e.InnerException;
+                }
+            }
+
+            throw new InvalidOperationException("No Convert method on type " + typeConverter.Name + " to convert from " +
+                typeSource.Name + " to " + typeDest.Name);
+        }
+
+        // Given a method info on the open generic; resolve it on a closed generic. 
+        // Beware that methods can be overloaded on name, so we can't just do a name match. 
+        // Nop if method is not generic. 
+        private static MethodInfo Resolve(Type type, MethodInfo method)
+        {
+            foreach (var candidate in type.GetMethods(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.Instance))
+            {
+                if (candidate.MetadataToken == method.MetadataToken)
+                {
+                    return candidate;                    
+                }
+            }
+            return null;
+        }
+
+        // Name can only map to a single type. If try to map to difference types, then it's a failed match. 
+        private static bool AddGenericArg(Dictionary<string, Type> genericArgs, string name, Type type)
+        {
+            Type typeExisting;
+            if (genericArgs.TryGetValue(name, out typeExisting))
+            {
+                return typeExisting == type;
+            }
+            genericArgs[name] = type;
+            return true;
+        }
+
+        // Return truen if the types are compatible. 
+        // If openType has generic args, then add a [Name,Type] entry to the genericArgs dictionary. 
+        private static bool CheckArg(Type openType, Type specificType, Dictionary<string, Type> genericArgs)
```

Comment:
There are a bunch of tricky private static helpers in this class that I think would benefit from some good unit tests, since they are so core. I looked quickly at the added tests, and didn't see any that seemed to exercise all the permutations here.